File: crud-react-webpage/src/models/Category.ts
/**
 * It represents category data
 */
export class Category{
    /**
     * Identifier of category
     */
    private id : string;
    /**
     * Name of category
     */
    private name : string;

    /**
     * Constructor from jsonObject
     * @param category jsonObject
     */
    constructor(category : any){
        if(category==undefined){
            this.id='';
            this.name='';
        }
        else{
            this.id = category.id;
            this.name = category.name;
        }        
    }

    /**
     * 
     * @returns identifier of category
     */
    public getId(): string {
        return this.id;
    }
    /**
     * 
     * @param value to set identifier of category
     */
    public setId(value: string) {
        this.id = value;
    }

    /**
     * 
     * @returns name of category
     */
    public getName(): string {
        return this.name;
    }
    /**
     * 
     * @param value to set name of category
     */
    public setName(value: string) {
        this.name = value;
    }
}
//export {Category}

File: spring-boot-webflux-api-rest/src/main/java/com/springboot/webflux/restapi/app/models/documents/Category.java
package com.springboot.webflux.restapi.app.models.documents;

import javax.validation.constraints.NotEmpty;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

/**
 * 
 * @author Mario Ruiz Rojo
 * It represents categories collection
 */
@Document(collection="categories")
public class Category {
	/**
	 * Identifier of category
	 */
	@Id
	@NotEmpty
	private String id;
	
	/**
	 * Name of category
	 */
	private String name;
	
	public Category() {

	}
	
	public Category(String name) {
		this.name = name;
	}

	/**
	 * 
	 * @return Identifier of category
	 */
	public String getId() {
		return id;
	}

	/**
	 * 
	 * @param id Identifier to set
	 */
	public void setId(String id) {
		this.id = id;
	}

	/**
	 * 
	 * @return Name of category
	 */
	public String getName() {
		return name;
	}

	/**
	 * 
	 * @param name to set in category
	 */
	public void setName(String name) {
		this.name = name;
	}
}
